bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPElevationPlotPlugin.java

```diff
@@ -234,98 +219,103 @@ private void _setColorCode(String label) {
-                SPNodeKey progKey = _selectedObservations[i].getProgramKey();
-                ISPProgram prog = db.lookupProgram(progKey);
-                // LORD OF DESTRUCTION: DataObjectManager get without set
-                SPProgram spProg = (SPProgram) prog.getDataObject();
-                queueBand = spProg.getQueueBand();
-                if (queueBand == null || queueBand.length() == 0)
-                    queueBand = "Default Band";
-                else
-                    queueBand = "Band " + queueBand;
-                if ((colors[i] = (Paint) paintMap.get(queueBand)) == null) {
-                    color = _COLORS[colorIndex++];
+                for (int i = 1; i < 5; i++) {
+                    color = _COLORS[colorIndex++ % _COLORS.length];
+                    queueBand = "Band " + i;
                     paintMap.put(queueBand, color);
-                    colors[i] = color;
                     lic.add(new LegendItem(queueBand, color));
                 }
-            }
-        } else if (_colorCodeTrajectories.equals(_COLOR_CODE_PRIORITY)) {
-            // get the order of items right
-            Paint color;
-            for (SPObservation.Priority pr: _PRIORITIES) {
-                color = _COLORS[colorIndex++];
-                paintMap.put(pr, color);
-                String label = pr.displayValue() + " Priority";
-                lic.add(new LegendItem(label, color));
-            }
-            for (int i = 0; i < _selectedObservations.length; i++) {
-                // LORD OF DESTRUCTION: DataObjectManager get without set
-                final SPObservation spObs = (SPObservation) _selectedObservations[i].getDataObject();
-                final SPObservation.Priority prio = spObs.getPriority();
-                if ((colors[i] = (Paint) paintMap.get(prio)) == null) {
-                    color = _COLORS[colorIndex++];
-                    paintMap.put(prio, color);
-                    colors[i] = color;
-                    String label = prio.displayValue() + " Priority";
-                    lic.add(new LegendItem(label, color));
+                IDBDatabaseService db = SPDB.get();
+                for (int i = 0; i < _selectedObservations.length; i++) {
+                    SPNodeKey progKey = _selectedObservations[i].getProgramKey();
+                    ISPProgram prog = db.lookupProgram(progKey);
+                    // LORD OF DESTRUCTION: DataObjectManager get without set
+                    SPProgram spProg = (SPProgram) prog.getDataObject();
+                    queueBand = spProg.getQueueBand();
+                    if (queueBand == null || queueBand.length() == 0)
+                        queueBand = "Default Band";
+                    else
+                        queueBand = "Band " + queueBand;
+                    if ((colors[i] = paintMap.get(queueBand)) == null) {
+                        color = _COLORS[colorIndex++ % _COLORS.length];
```

Comment:
Post-increment is a red flag that says we're doing something confusing, but it's not necessary here so I suggest switching to pre-increment. I may be over-stressed about this but indexing really gives me the heebiejeebies.

Reply:
Or at least open an issue to revisit this. It's not urgent.

Reply:
Alright. I did some refactoring of the code in `SPElevationPlotPlugin` as it was horrible, with assignments in `if` conditions (e.g. `if ((variable = newValue) != null)`), lots of repeated code, and as @tpolecat pointed out, array indexing that generated a red flag.

I think it is much nicer now, and much safer.

I realize that it is a substantial change, but I spent several hours testing it for various programs and selections of observations through all the different menu option combinations in the elevation plot against what we had in `2018A-test.1.4.1`, and the results were consistently identical as expected, so I'm hoping that despite the scope of the changes, we can get this merged.